clippy_lints/src/redundant_clone.rs

```diff
@@ -151,14 +165,13 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for RedundantClone {
                     }
                 };
 
-                let (local, cannot_move_out) = unwrap_or_continue!(find_stmt_assigns_to(
-                    cx,
-                    mir,
-                    pred_arg,
-                    true,
-                    mir[ps[0]].statements.iter()
-                ));
-                if cannot_move_out {
+                let (local, cannot_move_out) =
+                    unwrap_or_continue!(find_stmt_assigns_to(cx, mir, pred_arg, true, ps[0]));
+                let loc = mir::Location {
+                    block: bb,
+                    statement_index: mir.basic_blocks()[bb].statements.len(),
+                };
+                if cannot_move_out || possible_borrower.only_borrowers(&[arg, cloned][..], local, loc) != Some(true) {
```

Comment:
The only way `only_borrowers` is used is to compare with `Some(true)` and pass a slice. Please create a wrapper that does all this